framework/Models/Code.py

```diff
@@ -429,47 +429,50 @@ def _expandForWindows(self, origCommand):
     executable = commandSplit[0]
 
     if os.path.exists(executable):
-      with open(executable, "r+b") as executableFile:
-        firstTwoChars = executableFile.read(2)
-        if firstTwoChars == "#!":
-          realExecutable = shlex.split(executableFile.readline())
-          self.raiseAMessage("reading #! to find executable:" + repr(realExecutable))
-          # The below code should work, and would be better than findMsys,
-          # but it doesn't work.
-          # winExecutable = subprocess.check_output(['cygpath','-w',realExecutable[0]],shell=True).rstrip()
-          # print("winExecutable",winExecutable)
-          # realExecutable[0] = winExecutable
-          def findMsys():
-            """
-              Function to try and figure out where the MSYS64 is.
-              @ In, None
-              @ Out, dir, String, If not None, the directory where msys is.
-            """
-            dir = os.getcwd()
-            head, tail = os.path.split(dir)
-            while True:
-              if tail.lower().startswith("msys"):
-                return dir
-              dir = head
-              head, tail = os.path.split(dir)
-            return None
-          msysDir = findMsys()
-          if msysDir is not None:
-            beginExecutable = realExecutable[0]
-            if beginExecutable.startswith("/"):
-              beginExecutable = beginExecutable.lstrip("/")
-            winExecutable = os.path.join(msysDir, beginExecutable)
-            self.raiseAMessage("winExecutable " + winExecutable)
-            if not os.path.exists(winExecutable) and not os.path.exists(winExecutable + ".exe") and winExecutable.endswith("bash"):
-              #msys64 stores bash in /usr/bin/bash instead of /bin/bash, so try that
-              maybeWinExecutable = winExecutable.replace("bin/bash","usr/bin/bash")
-              if os.path.exists(maybeWinExecutable) or os.path.exists(maybeWinExecutable + ".exe"):
-                winExecutable = maybeWinExecutable
-            realExecutable[0] = winExecutable
-          else:
-            self.raiseAWarning("Could not find msys in "+os.getcwd())
-          commandSplit = realExecutable + [executable] + commandSplit[1:]
-          return commandSplit
+        try:
+          with open(executable, "r+b") as executableFile:
```

Comment:
You're suggesting we move this fix you wrote to initialization? Did you have problems with that when you first wrote this, or did it not come up as an option?

Reply:
No problem with this. I was providing my un-requested suggestion :). If you move this check at the initialize stage it will be done at the begin of the Step, making this issue almost impossible to happen (and would "speed up" the launching of the program since the check will be done once and not for each sample). The issue  could happen, in that case, only when multiple instances of RAVEN point to the same executable (E.g. RrR with inner using a Code).

Just my 2 cents :)

Reply:
Can you make the changes as requested by Andrea?

Reply:
I don't currently have the bandwidth to try this new approach with a variety of codes to see if it works; I don't have access to many of the codes we have interfaces with, and the `interfaceCheck` tests won't catch it. This will have to be closed and planned for another time if we want to do that level of change.